client/src/components/publications/Publications.js

```diff
@@ -1,54 +1,100 @@
     const dispatch = useDispatch();
     const { teamId } = useParams();
+    const allLayouts = {
+        allPublications: "All Publications",
+        byCategory: "By Category"
+    } 
     const [showCreateForm, setShowCreateForm] = useState(false)
     const [showImportForm, setShowImportForm] = useState(false)
+    const [layout, setLayout] = useState(allLayouts.allPublications)
 
     useEffect(() => {
         dispatch(getPublicationsByTeamId(teamId));
       }, [dispatch, teamId]);
 
     const teamPublications = useSelector(state => state.publications)
 
+    {/* Render publications based on the layout chosen */}
+    const renderPublications = () => {
+        if (layout === allLayouts.allPublications){
+            return(
+                <AllPublications teamPublications={teamPublications} />
+            )
+        } else if (layout === allLayouts.byCategory){
+            return(
+                <ByCategory teamPublications={teamPublications} />
+            )
+        }
+    }
+
     return (
         <> 
-            <div className="mb-3 mt-3 text-center">
-                <Button className="ml-2 mr-2" onClick={() => setShowCreateForm(true)}>    
-                    <IconContext.Provider value={{ color: 'black', size: '30px' }}>
-                        <VscAdd />
-                    </IconContext.Provider>
-                </Button>
-                <Button className="ml-2 mr-2"  onClick={() => setShowImportForm(true)}> 
-                    <IconContext.Provider value={{ color: 'black', size: '30px' }}>
-                        <BsArrowUpDown /> 
-                    </IconContext.Provider>
-                </Button>
-            </div>
+            <Container className="mt-4">
+                <Row>
+                    {/* Add publication and import publication button */}
+                    <Col md={{ span: 4, offset: 4 }}>
+                        <div className="mb-3 mt-3 text-center">
+                            <Button className="ml-2 mr-2" onClick={() => setShowCreateForm(true)}>    
+                                <IconContext.Provider value={{ color: 'black', size: '30px' }}>
+                                    <VscAdd />
+                                </IconContext.Provider>
+                            </Button>
+                            <Button className="ml-2 mr-2"  onClick={() => setShowImportForm(true)}> 
+                                <IconContext.Provider value={{ color: 'black', size: '30px' }}>
+                                    <BsArrowUpDown /> 
+                                </IconContext.Provider>
+                            </Button>
+                        </div>
+                    </Col>
+
+                    {/* Dropdown toggle for choosing different layout */}
+                    <Col md={{ span: 2, offset: 2}}>
+                        <div className="mb-3 mt-3 text-center">
+                            <h5 className="ml-2"> Choose layout: </h5>
```

Comment:
Here's a thought, instead of having choose layout as seperate text, why not have it in the button like "Layout: All Publications" or "Layout: By Category"?

Reply:
Is it what you mean? 
![image](https://user-images.githubusercontent.com/17608976/118258933-d051b380-b4f3-11eb-8435-375804ca61cf.png)

Reply:
Yup that's what I mean